arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceOptionsViewController.swift

```diff
@@ -0,0 +1,279 @@
+            let convertedValue: Any
+            
+            if let codedValue = value as? AGSCodedValue, attribute.domain is AGSCodedValueDomain {
+                // The value is a coded value.
+                convertedValue = convertToDataType(value: codedValue.code!, dataType: attribute.dataType)
+            } else {
+                // The value is from user input.
+                convertedValue = convertToDataType(value: value, dataType: attribute.dataType)
+            }
+            
+            if let expression = AGSUtilityNetworkAttributeComparison(networkAttribute: attribute, comparisonOperator: comparison, value: convertedValue) {
+                // Create and pass the valid expression back to the main view controller.
+                delegate?.optionsViewController(self, didCreate: expression)
+            }
+        }
+        dismiss(animated: true)
+    }
+    
+    @IBAction func cancelBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        dismiss(animated: true)
+    }
+    
+    // MARK: UI and data binding methods
+    
+    /// Convert the values to matching data types.
+    ///
+    /// - Parameters:
+    ///   - value: The right hand side value used in the conditional expression.
+    ///   - dataType: An `AGSUtilityNetworkAttributeDataType` enum case.
+    ///
+    /// - Note: The input value can either be an `AGSCodedValue` populated from the left hand side
+    ///         attribute's domain, or a numeric value entered by the user.
+    ///
+    /// - Returns: Converted value.
+    func convertToDataType(value: Any, dataType: AGSUtilityNetworkAttributeDataType) -> Any {
+        switch dataType {
+        case .integer:
+            return value as! Int64
+        case .float:
+            return value as! Float
+        case .double:
+            return value as! Double
+        case .boolean:
+            return value as! Bool
+        default:
+            return value
+        }
+    }
+    
+    func updateCellStates() {
+        // Disable the value cell when attribute is unspecified.
+        if let selectedAttribute = selectedAttribute {
+            if selectedAttribute.domain is AGSCodedValueDomain {
+                // Indicate that a new view controller will display.
+                valueCell.accessoryType = .disclosureIndicator
+            } else {
+                // Indicate that an alert will show.
+                valueCell.accessoryType = .none
+            }
+            valueCell.textLabel?.isEnabled = true
+            valueCell.isUserInteractionEnabled = true
+        } else {
+            // Enable the value cell when an attribute is specified.
+            valueCell.textLabel?.isEnabled = false
+            valueCell.isUserInteractionEnabled = false
+        }
+    }
+    
+    // Transition to the attribute options view controller.
+    func showAttributePicker() {
+        let selectedIndex = possibleAttributes.firstIndex { $0 == selectedAttribute } ?? -1
```

Comment:
I didn't realize that `-1` is used as the selected index in some cases, which currently isn't supported by `OptionsTableViewController`. Due to recent changes to that class, passing `-1` as the selected index leads to a crash when a row is selected. I'll submit a PR to address that, but it will mean that this branch will need to be updated to account for those changes.

Reply:
Nice catch. I did a quick search in the project with `selectedIndex:` and it seems the existing samples doesn't use -1 to nilify the selection. After #962 is merged, we can fix it with a nil here.